### PR TITLE
docs: addition of param and result sections

### PR DESCRIPTION
**Pull request by hachreak**
* Improves documentation. (addresses #137)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio_accounts/ext.py [hunk 1]
```diff
@@ -50,20 +50,33 @@
...
+    """
     return _to_binary(password)
 
 
 def encrypt_password(password):
-    """Override Flask-Security's default encryption function."""
+    """Override Flask-Security's default encryption function.
+
+    :param password: The plain password.
+    :returns: The password encrypted.
```
Comment (alizeepace) [inline invenio_accounts/ext.py]: The encrypted password

File: invenio_accounts/ext.py [hunk 2]
```diff
@@ -84,9 +97,27 @@ def init_app(self, app, use_celery=True, sessionstore=None,
                  register_blueprint=True):
         """Flask application initialization.
 
+        The following actions are executed:
+
+        #. Initialized the configuration.
```
Comment (alizeepace) [inline invenio_accounts/ext.py]: I would put all the following verbs either in the infinite form (Initialize, Monkey-patch, Create, confirm) like "register" line 112, or in the third-person present form (Initializes, Monkey-patches, Creates, registers) like "confirms" line 111.

File: invenio_accounts/ext.py [hunk 3]
```diff
@@ -211,10 +245,12 @@ def init_app(self, app, use_celery=True, sessionstore=None,
...
         """Flask application initialization.
 
+        :param app: The Flask application.
         :param sessionstore: store for sessions. Passed to
-            ``flask-kvsession``. Defaults to redis.
-        :param register_blueprint: Do not register the Security blueprint
-            by default.
+            ``flask-kvsession``. If ``None`` then Redis is configured.
+            (Default: ``None``)
+        :param register_blueprint: If ``True``, the application register the
```
Comment (alizeepace) [inline invenio_accounts/ext.py]: registers

File: invenio_accounts/config.py [hunk 4]
```diff
@@ -24,10 +24,16 @@
...
+
+It should be in the form ``redis://username:password@host:port/db_index``.
+"""
 
-# ACCOUNTS_FORCE_BLUEPRINT_REGISTRATION can be use to override
-# ``register_blueprint`` option.
 ACCOUNTS_REGISTER_BLUEPRINT = None
+"""Register the Security blueprint or not.
+
+It can be use to override the ``register_blueprint`` option.
```
Comment (alizeepace) [inline invenio_accounts/config.py]: used to

File: invenio_accounts/ext.py [hunk 5]
```diff
@@ -80,13 +93,29 @@ def monkey_patch_flask_security():
...
+
+        #. Initialize the configuration.
+
+        #. Monkey-patch Flask-Security.
+
+        #. Create the user datastore.
+
+        #. Create the sessionstore.
+
+        #. Initialize the extension, the forms for register users and
```
Comment (alizeepace) [inline invenio_accounts/ext.py]: the forms to register and confirm users ?
or: to register users and send them a confirmation
(not sure about the meaning of confirm in that context)

File: invenio_accounts/ext.py [hunk 6]
```diff
@@ -80,13 +93,29 @@ def monkey_patch_flask_security():
...
+            confirms them, the CLI and, if ``ACCOUNTS_USE_CELERY`` is ``True``,
+            register a celery task to send emails.
+
+        :param app: The Flask application.
         :param sessionstore: store for sessions. Passed to
-            ``flask-kvsession``. Defaults to redis.
-        :param register_blueprint: Register the Security blueprint or not.
+            ``flask-kvsession``. If ``None`` then Redis is configured.
+            (Default: ``None``)
+        :param register_blueprint: If ``True``, the application register the
```
Comment (alizeepace) [inline invenio_accounts/ext.py]: registers

File: invenio_accounts/hash.py [hunk 7]
```diff
@@ -61,26 +61,42 @@ def _mysql_aes_key(key):
...
 
 def mysql_aes_encrypt(val, key):
-    """Mysql AES encrypt value with secret key."""
+    """Mysql AES encrypt value with secret key.
+
+    .. note:: Useful only for MySQL.
+
+    :param val: Plain text value.
+    :param key: The AES key.
+    :returns: The AES value encrypted.
```
Comment (alizeepace) [inline invenio_accounts/hash.py]: The encrypted AES value. (see also line 116)

File: invenio_accounts/models.py [hunk 8]
```diff
@@ -58,25 +61,35 @@ class User(db.Model, UserMixin):
...
+    """User email."""
 
     password = db.Column(db.String(255))
+    """User password."""
 
     active = db.Column(db.Boolean)
+    """Flag to say if the user is active or not ."""
 
     confirmed_at = db.Column(db.DateTime)
+    """When the user confirm the email address."""
```
Comment (alizeepace) [inline invenio_accounts/models.py]: confirmed

File: invenio_accounts/models.py [hunk 9]
```diff
@@ -58,25 +61,35 @@ class User(db.Model, UserMixin):
...
 
     last_login_at = db.Column(db.DateTime)
+    """When the user logged in last time."""
```
Comment (alizeepace) [inline invenio_accounts/models.py]: for the last time

File: invenio_accounts/models.py [hunk 10]
```diff
@@ -58,25 +61,35 @@ class User(db.Model, UserMixin):
...
 
     current_login_at = db.Column(db.DateTime)
+    """When user logged in."""
```
Comment (alizeepace) [inline invenio_accounts/models.py]: When the user logged in.
(or you could maybe use 'logged into the current session' for more clarity)

File: invenio_accounts/models.py [hunk 11]
```diff
@@ -58,25 +61,35 @@ class User(db.Model, UserMixin):
...
 
     current_login_ip = db.Column(IPAddressType, nullable=True)
+    """Current user IP address."""
 
     login_count = db.Column(db.Integer)
+    """Count how many times the user logged in."""
 
     roles = db.relationship('Role', secondary=userrole,
                             backref=db.backref('users', lazy='dynamic'))
+    """List of user's roles."""
```
Comment (alizeepace) [inline invenio_accounts/models.py]: of the user's

**Comment by hachreak** [comment]
ping @jirikuncar

**Comment by hachreak** [comment]
ping @jirikuncar

File: invenio_accounts/hash.py [hunk 12]
```diff
@@ -61,26 +61,42 @@ def _mysql_aes_key(key):
 
 
 def _mysql_aes_pad(val):
-    """Padding."""
+    """Padding.
+
+    .. note:: Useful only for MySQL.
```
Comment (jirikuncar) [inline invenio_accounts/hash.py]: @hachreak the note is obvious from the function name. Can you remove it for functions prefixed with `_?mysql`?

**Comment by jirikuncar** [comment]
@hachreak LGTM

**Comment by hachreak** [comment]
@jirikuncar :shipit: ?

**Pull request merged**